src/python/pants/backend/terraform/dependency_inference.py

```diff
@@ -139,9 +143,45 @@ async def infer_terraform_module_dependencies(
     return InferredDependencies(terraform_module_addresses)
 
 
+@dataclass(frozen=True)
+class TerraformDeploymentDependenciesInferenceFieldSet(TerraformDeploymentFieldSet):
+    pass
+
+
+class InferTerraformDeploymentDependenciesRequest(InferDependenciesRequest):
+    infer_from = TerraformDeploymentDependenciesInferenceFieldSet
+
+
+@rule
+async def infer_terraform_deployment_dependencies(
+    request: InferTerraformDeploymentDependenciesRequest,
+) -> InferredDependencies:
+    root_module_address_input = request.field_set.root_module.to_address_input()
+    root_module = await Get(Address, AddressInput, root_module_address_input)
+
+    # # TODO: This leads to a dependency on self. Can we do something about it?
```

Comment:
Why does that happen? I don't see it off hand

Reply:
ah, right, this was a bit of a bodge past that. The problem is what to put for the "target_name". You can't just make something up, but if you use the current target then it depends on self. I've updated the comment.